### PR TITLE
[CALCITE] Added token assignment parsing logic to gradle task

### PR DESCRIPTION
**Pull request by AndrewPochapsky**
- Added regex pattern for parsing statements like \[\<OPT1, OPT2, ...\>\](TOKEN|SKIP|MORE) :
- Created class ExtractedData which holds extracted functions and extracted token assignments
- Refactored processFile() to allow for two passes of the file text, one for functions and a second for token assignments
- Added tests for parsing token assignments

### CODE REVIEW

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java [hunk 1]
```diff
@@ -56,12 +57,13 @@ public DialectTraverser(File dialectDirectory, File rootDirectory,
...
-   * Extracts functions and prints the results for the given dialect.
+   * Extracts functions and token assignments and prints the functions for the
+   * given dialect.
    */
   public void run() {
-    Map<String, String> functions = extractFunctions();
+    ExtractedData extractedData = extractData();
     // TODO(AndrewPochapsky): Remove this once generation logic added.
-    for (Map.Entry<String, String> entry : functions.entrySet()) {
+    for (Map.Entry<String, String> entry : extractedData.functions.entrySet()) {
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java:66]: Should we also print out the tokens assignments?

File: buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java [hunk 2]
```diff
@@ -177,4 +189,29 @@ private String readFile(Path path) {
   @Test public void processFileMultipleFunctionsSeparatedByLines() {
     assertFileProcessed("multiple_functions_separated");
   }
+
+  @Test public void processFileTokenAssignments() {
+    assertFileProcessed("token_assignments");
+  }
+
+  @Test public void processFileFunctionsAndAssignments() {
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java:197]: nit: rename to processFileFunctionsAndTokenAssignments?

File: buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java [hunk 3]
```diff
@@ -177,4 +189,29 @@ private String readFile(Path path) {
...
+  }
+
+  @Test public void processTokenAssignmentEmpty() {
+    String declaration = "TOKEN :\n";
+    String assignment = declaration
+      + "{\n}";
+    assertTokenAssignmentIsParsed(declaration, assignment);
+  }
+
+  @Test public void processTokenAssignmentNonEmpty() {
```
Comment (dasch-1) [inline buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java:208]: Should we add unit test cases for MORE and SKIP token assignments?

File: buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java (hunk 1 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java:66]: Sure, will add.

File: buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java (hunk 3 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java:208]: Will add.

File: buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java (hunk 2 above)
Comment (AndrewPochapsky) [reply buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java:197]: Sure.

**Pull request merged**